src/sql/workbench/browser/modal/modal.ts

```diff
@@ -316,29 +317,30 @@ export abstract class Modal extends Disposable implements IThemable {
 	}
 
 	/**
-	 * Set focusable elements in the modal dialog
+	 * Figures out the first and last elements which the user can tab to in the dialog
 	 */
-	public setFocusableElements() {
-		// try to find focusable element in dialog pane rather than overall container
-		this._focusableElements = this._modalBodySection ?
-			this._modalBodySection.querySelectorAll('a[href], area[href], input:not([disabled]), select:not([disabled]), textarea:not([disabled]), button:not([disabled]), [tabindex="0"]') :
-			this._bodyContainer.querySelectorAll('a[href], area[href], input:not([disabled]), select:not([disabled]), textarea:not([disabled]), button:not([disabled]), [tabindex="0"]');
-		if (this._focusableElements && this._focusableElements.length > 0) {
-			this._firstFocusableElement = <HTMLElement>this._focusableElements[0];
-			this._lastFocusableElement = <HTMLElement>this._focusableElements[this._focusableElements.length - 1];
+	public setFirstLastTabbableElement() {
+		let tabbableElements = this._bodyContainer.querySelectorAll('a[href], area[href], input:not([disabled]), select:not([disabled]), textarea:not([disabled]), button:not([disabled]), [tabindex="0"]');
+		if (tabbableElements && tabbableElements.length > 0) {
+			this._firstTabbableElement = <HTMLElement>tabbableElements[0];
+			this._lastTabbableElement = <HTMLElement>tabbableElements[tabbableElements.length - 1];
 		}
-
-		this._focusedElementBeforeOpen = <HTMLElement>document.activeElement;
-		this.focus();
 	}
 
 	/**
-	 * Focuses the modal
-	 * Default behavior: focus the first focusable element
+	 * Set focusable elements in the modal dialog
 	 */
-	protected focus() {
-		if (this._firstFocusableElement) {
-			this._firstFocusableElement.focus();
+	public setFocusableElements() {
```

Comment:
This name could probably be updated now since it's really just initializing the first focused element on open. Maybe something like `setInitialFocusedElement()`

Reply:
That is a better name now. I'll update it.